### ISSUE TITLE
comparison of DE outputs pre/post pseudobulking

### ISSUE THREAD

**Issue by antoine4ucsd** [satijalab/seurat#8341]
Hello
I am analyzing a set of scRNAseq integrated with Harmony
my goal is to compare 2 groups of samples but trying to figure out the best approach...
Following this nice vignette
https://satijalab.org/seurat/articles/de_vignette

#1 without pseudobulking trying the following
```
data1<- FindMarkers(
        object = data,
        ident.1 = "grp1",
        ident.2 = "grp2",
        assay = "SCT")
data2<- FindMarkers(
        object = data,
        ident.1 = "grp1",
        ident.2 = "grp2",
        assay = "RNA")
```

resulting in these 
![Rplot](https://github.com/satijalab/seurat/assets/18502631/2c00e6e4-43bc-47aa-96a9-0aa64d636cce)

#2 DESEQ2 test does not work without pseudolbuk
```
data3 <- FindMarkers(
        object = data,
        ident.1 = "grp1",
        ident.2 = "grp2",
        assay = "RNA",
test.use = "DESeq2")
```
error
converting counts to integer mode
Error in estimateSizeFactorsForMatrix(counts(object), locfunc = locfunc,  : 
  every gene contains at least one zero, cannot compute log geometric means

#3 Next I tried after aggregation / pseudobulking
```
dataagg_SCT <- AggregateExpression(data, assays = "SCT", return.seurat = T, group.by = c( "pid", "status"))
data3 <- FindMarkers(object = dataagg_SCT,
                                assay = "SCT",
                         ident.1 = "grp1",
                         ident.2 = "grp2",
                         test.use = "DESeq2")

dataagg_RNA <- AggregateExpression(data, assays = "RNA", return.seurat = T, group.by = c( "pid", "status"))
data4  <- FindMarkers(object = dataagg_RNA,
                                assay = "RNA",
                         ident.1 = "grp1",
                         ident.2 = "grp2",
                         test.use = "DESeq2")
```
resulting in these . the results are totally different as you can see and shifted toward grp1 (not centered to zero), which makes me believe I am missing a rescaling step? Wilcoxon test gives no meaningful results

![Rplot012](https://github.com/satijalab/seurat/assets/18502631/0ea21496-69c0-4e1e-8603-8304c2bc9bc9)

All suggestions are very welcome to help refining these analyses. I was curious about getting DESEQ2 results but does not work without pseudobulking on my data

thank you in advance for your help

**Comment by camara-h** [comment]
Hi @antoine4ucsd, thanks for bringing that up.
Maybe I don`t have much to add in the troubleshooting part, but we are running into similar issues with our datasets.

I tried comparisons with three different datasets and would show two of them:

1.
`pseudo_s.obj <- AggregateExpression(s.object, assays = "RNA", slot = "counts", return.seurat = T, group.by = c("cell_type_res.1.6", "sample", "neck_region"))
pseudo_s.obj$cell_type_res.1.6.neck_region <- paste(pseudo_s.obj$cell_type_res.1.6, pseudo_s.obj$neck_region, sep = "_")
bulk.mono.de <- FindMarkers(object = pseudo_s.obj, 
                         ident.1 = "White-adipocyte-2_Deep", 
                         ident.2 = "White-adipocyte-2_Superficial",
                         test.use = "DESeq2")
 `
![image](https://github.com/satijalab/seurat/assets/72557747/1dc513c7-dd63-492c-b0b2-b7cfe119a128)


2.
`pseudo_s.obj <- AggregateExpression(seurat_NIC, assays = "RNA", slot = "counts", return.seurat = T, group.by = c("integrated_snn_res.1.8", "orig.ident", "sample"))
pseudo_s.obj$cluster.sample <- paste(pseudo_s.obj$integrated_snn_res.1.8, pseudo_s.obj$sample, sep = "_")
bulk.mono.de <- FindMarkers(object = pseudo_s.obj, 
                         ident.1 = "g28", 
                         ident.2 = "g1",
                         test.use = "DESeq2")
`
![image](https://github.com/satijalab/seurat/assets/72557747/02db2640-16fc-424f-8710-387671804a4b)

Hope it`s anything helpful :)

**Comment by longmanz** [comment]
Hi @antoine4ucsd ,

I suspect this is caused by the number of pseudo-bulked "cells" being too small. How many "cells "are there in your "grp1" and "grp2" group?

**Comment by antoine4ucsd** [comment]
you are probably right. I have 4 samples in grp1 and 3 samples in grp2. Maybe avoid pseudo-bulking in that case? 
thank you

**Comment by longmanz** [comment]
Hi, 
We have fixed this logFC issue in our v5.0.2 and subsequent versions (by implementing a normalization step when calculating the logFC even if the 'counts' slot is used for DE testing. see https://github.com/satijalab/seurat/releases for details). Thank you for reporting this.
